src/modules/form/components/viewable/DynamicViewField.tsx

```diff
@@ -0,0 +1,189 @@
+    fetchPolicy: 'network-only', // Always fetch, because ProjectCoC and Enrollment records change
+    onCompleted: (data) => {
+      if (!data?.pickList) return;
+      if (!hasMeaningfulValue(value)) return;
+
+      // Try to find the "full" option (including label) for this value from the pick list
+      let fullOption;
+      if (isPickListOption(value)) {
+        fullOption = data.pickList.find((o) => o.code === value.code);
+      } else if (isPickListOptionArray(value)) {
+        fullOption = intersectionBy(data.pickList, value, 'code');
+      }
+
+      if (fullOption) {
+        // Update the value so that it shows the complete label
+        adjustValue({ linkId: item.linkId, value: fullOption });
+      } else {
+        console.warn(
+          `Selected value '${JSON.stringify(
+            value
+          )}' is not present in option list '${item.pickListReference}'`
+        );
+      }
+    },
+  });
+
+  const commonProps = useMemo(
+    () => ({
+      label,
+      value,
+      horizontal,
+    }),
+    [label, value, horizontal]
+  );
+
+  switch (item.type) {
+    case ItemType.Display:
+      return <DynamicDisplay item={item} />;
+    case ItemType.Boolean:
+      return (
+        <TextContent
+          {...commonProps}
+          value={
+            [TRUE_OPT, FALSE_OPT].find((o) => o.code === String(value))
+              ?.label || <NotCollectedText variant='body2' />
+          }
+          hasValue={(val) => !isNil(val)}
+        />
+      );
+    case ItemType.String:
+    case ItemType.Text:
+      return <TextContent {...commonProps} />;
+    case ItemType.Integer:
+      return <TextContent {...commonProps} />;
+    case ItemType.Currency:
+      return (
+        <TextContent
+          {...commonProps}
+          hasValue={(val) => !isNil(val)}
+          renderValue={(val) => `$${val}`}
+        />
+      );
+    case ItemType.Date:
+      return (
+        <TextContent
+          {...commonProps}
+          renderValue={(val) => {
+            let formatted;
+            if (val instanceof Date) formatted = format(val, 'M/d/yyyy');
+            if (typeof val === 'string') {
+              const parsed = parseHmisDateString(val);
+              if (parsed) formatted = format(parsed, 'M/d/yyyy');
+            }
+            if (formatted) return formatted;
+            return (
+              <Typography color='textSecondary' variant='body2'>
+                Invalid Date: {String(val)}
+              </Typography>
+            );
+          }}
+        />
+      );
+    case ItemType.OpenChoice:
+      return (
+        <TextContent
+          {...commonProps}
+          hasValue={(val) => hasMeaningfulValue(val)}
+          renderValue={(val) => {
+            if (pickListLoading) return <Skeleton>{val}</Skeleton>;
+            return (
+              <Typography variant='body2'>
+                {ensureArray(value)
+                  .map((val) => val.code)
+                  .join(', ')}
```

Comment:
```suggestion
                {ensureArray(value)
                  .map((val) => val.label || val.code)
                  .join(', ')}
```

And I don't see why `ItemType.Choice` can't have the same render function. Actually we should be able to collapse `case ItemType.OpenChoice` and `case ItemType.Choice`.